cnd/src/http_api/swap_resource.rs

```diff
@@ -107,21 +108,48 @@ pub enum IncludeState {
     No,
 }
 
+/// Build siren entity.  For failed swaps do not return any actions.
+pub fn build_rfc003_siren_entity_no_err<S: StateStore>(
+    state_store: &S,
+    swap: Swap,
+    types: SwapTypes,
+    include_state: IncludeState,
+) -> anyhow::Result<siren::Entity> {
+    build_rfc003_siren_entity(state_store, swap, types, include_state, false)
```

Comment:
Ok, here is a proposal that would make things clearer (I believe):
1. Do not add these new wrapper functions `build_rfc003_siren_entity_no_err`/`build_rfc003_siren_entity` and directly use `build_rfc003_siren_entity`
1. Instead of passing a bool to `build_rfc003_siren_entity` pass a new enum:
```
pub enum OnFail {
   Error,
   NoAction
}
```

I think it would make things clearer than having to comment and append `_no_err` and use wrapper and having bool argument.

Reply:
I'd definitely upvote for enum, way more readable than a random boolean.

Reply:
Cool, nice design.  Will implement.  This is Rust not C right :)

Reply:
I did not want to comment but I did feel this was coming from a C dev.